docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
++nhmacaddrs=(40:a6:b7:78:3f:bc 40:a6:b7:78:3f:b8)
+```
+
+6) Modify the script [docker_setup.sh](./scripts/docker_setup.sh) and update the function `move_ifaces()` in condition `if [ "$mode" == 'cndp' ]` . Update `start_q_idx` to choose the start queue index to receive n/w packets. This should match the queue id used in lports section of [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc).To get better performance (optional step), assign `cpuset-cpus` in [docker_setup.sh](./scripts/docker_setup.sh) to cores ids same as queue ids used to receive n/w packets.
+
+```
+diff --git a/conf/cndp_upf_1worker.jsonc b/conf/cndp_upf_1worker.jsonc
+index da60d51..b1c3df6 100644
+--- a/conf/cndp_upf_1worker.jsonc
++++ b/conf/cndp_upf_1worker.jsonc
+@@ -86,7 +86,7 @@
+     "lports": {
+         "enp134s0:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 6,
+             "umem": "umem0",
+             "region": 0,
+             "busy_poll": true,
+@@ -96,7 +96,7 @@
+         },
+         "enp136s0:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 6,
+             "umem": "umem0",
+             "region": 1,
+             "busy_poll": true,
+```
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 9058839..2e4e505 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -104,7 +104,7 @@ function move_ifaces() {
+                        # num queues
+                        num_q=1
+                        # start queue index
+-                       start_q_idx=22
++                       start_q_idx=6
+                        # RSS using TC filter
+                        setup_tc "${ifaces[$i]}" $num_q $start_q_idx
+                fi
+@@ -218,7 +218,7 @@ fi
+ # Run bessd
+ docker run --name bess -td --restart unless-stopped \
+-       --cpuset-cpus=12-13 \
++       --cpuset-cpus=6-7 \
+        --ulimit memlock=-1 -v /dev/hugepages:/dev/hugepages \
+        -v "$PWD/conf":/opt/bess/bessctl/conf \
+        --net container:pause \
+
+```
+
+7) Modify the script [cndp_reset_upf.sh](./scripts/cndp_reset_upf.sh) to use appropriate PCIe device address, network interface name and set_irq_affinity script in NIC driver.
+
+```
+ACCESS_PCIE=0000:86:00.0
+CORE_PCIE=0000:88:00.0
+ACCESS_IFACE=enp134s0
+CORE_IFACE=enp136s0
+SET_IRQ_AFFINITY=~/nic/driver/ice-1.9.7/scripts/set_irq_affinity
+```
+
+This script is used to stop any running containers, disable irqbalance, set irq affinity to all queues for access and core interface. The script also set XDP socket busy poll settings for access and core interfaces. `set_irq_affinity` script used by this script can be found in the NIC driver install path. irq affinity and AF_XDP busypoll settings are done to get improved network I/O performance. These settings are recommended but not mandatory.
+
+8) From the top level directory call:
+
+```
+$ ./scripts/cndp_reset_upf.sh
```

Comment:
This script uses `dpdk-devbind.py`. So, somewhere in the instructions you may need to indicate that the user needs to download this script from the DPDK repository. You can add some details similar to what was added in the instructions to run the UPF in DPDK mode (https://github.com/omec-project/upf/blob/master/docs/dpdk-configuration.md?plain=1#L34)

Reply:
ok